### ISSUE TITLE
smart beginning of line feature request

### ISSUE THREAD

**Issue by deadcyclo** [atom/autocomplete-plus#996]
### Prerequisites

* [X ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description
Would it be possible to ask you to implement a "smart beginning of line" (which is normally the first thing anybody sensible puts in their emacs config). 

Basically the possibility of having ctrl-a jump to the beginning of the text on the line, then when pressing ctrl-a again it jumps to the beginning of the line. Subsequent presses continue to jump back and forth between the two. 

This is the one major feature I miss in atom.

This could easily be a toggleable feature.

**Comment by Aerijo** [comment]
@deadcyclo I'm not sure what you're asking for here: if a line is indented, pressing <kbd>ctrl-a</kbd> will jump to the first non-whitespace character, pressing again will jump to the start of the line, and repeated presses will jump back and forth.

This seems to be exactly what you described. Also, as keybindings are dependent on OS, the command I'm referring to is `Editor: Move To First Character Of Line` in the command palette. 

If I've misunderstood, or this is not the case for you, please reopen with the template filled out complete with example text that doesn't work and steps to reproduce.
____
Thanks for reaching out!

We require the template to be filled out on all new issues and pull requests. We do this so that we can be certain we have all the information we need to address your submission efficiently. This allows the maintainers to spend more time fixing bugs, implementing enhancements, and reviewing and merging pull requests.

Thanks for understanding and meeting us half way :grinning:
